labgrid/logging.py

```diff
@@ -0,0 +1,262 @@
+import re
+import logging
+
+import attr
+
+from .step import steps, StepEvent
+
+DEFAULT_FORMAT = "%(levelname)-7.7s %(name)15.15s: %(message)s"
+
+def basicConfig(**kwargs):
+    kwargs.setdefault("format", DEFAULT_FORMAT)
+    indent = kwargs.get("indent", True)
+    logging.basicConfig(**kwargs)
+    root = logging.getLogger()
+
+    parent = root.handlers[0].formatter
+    root.handlers[0].setFormatter(StepFormatter(indent=indent, parent=parent))
+
+
+logging.CONSOLE = logging.INFO - 5
+assert(logging.CONSOLE > logging.DEBUG)
+logging.addLevelName(logging.CONSOLE, "CONSOLE")
+
+# Use composition instead of inheritance
+class StepFormatter:
+    def __init__(
+        self, *args, indent=True, color=None, parent=None, **kwargs
+    ):
+        if parent:
+            self.formatter = parent
+        else:
+            self.formatter = logging.Formatter(*args, **kwargs)
+        self.indent = indent
+        self.indent_level = 0
+        self.bufs = dict()
+        self.re_vt100 = re.compile(r"(\x1b\[|\x9b)[^@-_a-z]*[@-_a-z]|\x1b[@-_a-z]")
```

Comment:
The same regex is used in the BareboxDriver, ShellDriver and UBootDriver. Maybe store it somewhere in util and use it from all of these drivers, including logging?